eval: remove wrapping from builtin errors

yuzefovich opened the pull request:
This commit removes the code to perform error wrapping around the errors produced by the builtin function which provides a bit more context (like the function name). This wrapping dates all the way back to a commit from 2015 716837eb8a9385ccde2292ae4f1465235a4ddc93. Over the years it was expanded to also provide some more context to the sentry reports (7b703dede2459ed9ccef4288d403dac18701eb68). I don't think either of these provide much value since it's usually clear which builtin the error corresponds to and having just the builtin name in the sentry report doesn't make it any more actionable. Removing of this logic allows us to match errors as they are produced by Postgres.

Epic: None
Release note: None

yuzefovich committed:
eval: remove wrapping from builtin errors

This commit removes the code to perform error wrapping around the errors
produced by the builtin function which provides a bit more context (like
the function name). This wrapping dates all the way back to a commit
from 2015 716837eb8a9385ccde2292ae4f1465235a4ddc93. Over the years it
was expanded to also provide some more context to the sentry reports
(7b703dede2459ed9ccef4288d403dac18701eb68). I don't think either of
these provide much value since it's usually clear which builtin the
error corresponds to and having just the builtin name in the sentry
report doesn't make it any more actionable. Removing of this logic
allows us to match errors as they are produced by Postgres.

Release note: None

yuzefovich committed:
WIP on test adjustment

This commit removes the builtin context from error messages that was
removed in the previous commit. It'll be squashed into that one.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/143950)
<!-- Reviewable:end -->

yuzefovich committed:
WIP on pg_regress updates

yuzefovich committed:
WIP on TestInternalErrorReporting adjustment

This commit extracts the update to `TestInternalErrorReporting` as
a result of the change in the first commit (we no longer have any
special handling for builtin errors in sentry). It'll be squashed into
the first commit.
